### PR TITLE
Refactor parser script

### PR DESCRIPTION
**Pull request by NathanLovato**
Removed global statement, simplified the parse function.
Added docstrings.
Made the code a little easier to read.

I did that because I didn't understand the parser code the first time I read it. This preserves the tests' performances while hopefully improving readability.

### CODE REVIEW

File: gdtoolkit/parser.py [hunk 1]
```diff
@@ -39,8 +13,33 @@ class Indenter(indenter.Indenter):
...
+    parser="lalr",
+    start="start",
+    propagate_positions=True,
+)
+_PARSER: Tree = Lark.open(
+    os.path.join(THIS_DIRECTORY, "gdscript.lark"),
+    postlex=Indenter(),
+    parser="lalr",
+    start="start",
+)
```
Comment (Scony) [inline gdtoolkit/parser.py]: the caching must be done lazily as `Lark.open()` is time-consuming. Having those two loaded upfront will make e.g. `gdlint` loading `_PARSER` which is actually not used by linter. So in other words, the `gdlint` will call `Lark.open()` two times, hence causing the slowdown.

Comment (NathanLovato) [reply gdtoolkit/parser.py]: Sure. I'll try to refactor it in a class with cached and lazily evaluated properties.

**Comment by NathanLovato** [comment]
How about this code? I'm using a `cached_property` decorator and a class for delayed evaluation - and evaluating the parsers only once.

This feature is in the standard library but only from Python 3.8. For compatibility with py 3.7, you need the `cached_property` class.

**Comment by NathanLovato** [comment]
I also formatted the python code in bin/ with black

**Pull request merged**